sycl/include/CL/sycl/detail/context_impl.hpp

```diff
@@ -58,14 +58,22 @@ class context_impl {
   RT::PiContext &getHandleRef();
   const RT::PiContext &getHandleRef() const;
 
+  // Operating the returned map won't be thread-safe,
+  // use getProgramKernelCacheMutex() along with this method
```

Comment:
rather than exposing the mutex itself I think it would be better to just add locking semantics into the methods for working with the cache, e.g. acquire/release the cache.

Reply:
Employed unique_lock for this.